spec/datetimeformat.html

```diff
@@ -286,43 +273,517 @@ <h1>Internal slots</h1>
+              <td>Optional field. Present if [[pattern]] contains the substring *"{minute}"*.</td>
+            </tr>
+            <tr>
+              <td>[[second]]</td>
+              <td>[[Second]] values in the Values column of <emu-xref href="#table-datetimeformat-components"></emu-xref></td>
+              <td>Optional field. Present if [[pattern]] contains the substring *"{second}"*.</td>
+            </tr>
+            <tr>
+              <td>[[fractionalSecondDigits]]</td>
+              <td>[[FractionalSecondDigits]] values in the Values column of <emu-xref href="#table-datetimeformat-components"></emu-xref></td>
+              <td>Optional field. Present if [[pattern]] contains the substring *"{fractionalSecondDigits}"*.</td>
+            </tr>
+            <tr>
+              <td>[[timeZoneName]]</td>
+              <td>[[TimeZoneName]] values in the Values column of <emu-xref href="#table-datetimeformat-components"></emu-xref></td>
+              <td>Optional field. Present if [[pattern]] contains the substring *"{timeZoneName}"*.</td>
+            </tr>
+            <tr>
+              <td>[[pattern]]</td>
+              <td>a Pattern String</td>
+              <td><emu-not-ref>Contains</emu-not-ref> for each of the date and time format component fields of the record a substring starting with *"{"*, followed by the name of the field, followed by *"}"*. If the record has a [[year]] field, the string may contain the substrings *"{yearName}"* and *"{relatedYear}"*.</td>
+            </tr>
+            <tr>
+              <td>[[pattern12]]</td>
+              <td>a Pattern String</td>
+              <td>Optional field. Present if the [[hour]] field is present. In addition to the substrings of the [[pattern]] field, contains at least one of the substrings *"{ampm}"* or *"{dayPeriod}"*.</td>
+            </tr>
+            <tr>
+              <td>[[rangePatterns]]</td>
+              <td>a DateTime Range Pattern Record</td>
+              <td>Pattern strings in this field are similar to [[pattern]].</td>
+            </tr>
+            <tr>
+              <td>[[rangePatterns12]]</td>
+              <td>a DateTime Range Pattern Record</td>
+              <td>Optional field. Present if the [[hour]] field is present. Pattern strings in this field are similar to [[pattern12]].</td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-datetimeformat-range-pattern-record">
+        <h1>DateTime Range Pattern Records</h1>
+
+        <p>
+          Each <dfn id="datetimeformat-range-pattern-record">DateTime Range Pattern Record</dfn> has the fields defined in <emu-xref href="#table-datetimeformat-range-pattern-record"></emu-xref>.
+        </p>
+        <emu-table id="table-datetimeformat-range-pattern-record">
+          <emu-caption>DateTime Range Pattern Record</emu-caption>
```

Comment:
I have a goal of replacing each [[rangePatterns]] and [[rangePatterns12]] unit-keyed Record with a List of Records with fields like { [[BiggestDivergentUnit]]: string, [[RangePattern]]: DateTimeFormat Range Pattern Record } Records (or possibly just a list of range pattern Records that include that first field, saving a level of depth), and it might be appropriate to include that change in this PR rather than introduce new bibliographic terms that will just need to be removed. Thoughts?

<details>

The net effect should be to reduce opacity of field semantics, replacing e.g.
```
[[rangePatterns]]: {
  [[Hour]]: {
    [[hour]]: "numeric",
    [[minute]]: "numeric",
    [[PatternParts]]: […]
  },
  [[Minute]]: {
    [[hour]]: "numeric",
    [[minute]]: "numeric",
    [[PatternParts]]: […]
  },
  [[Default]]: {
    [[year]]: "2-digit",
    [[month]]: "numeric",
    …,
    [[PatternParts]]: […]
}
```
with
```
[[rangePatterns]]: [
  { [[BiggestDivergentUnit]]: "hour",
    [[hour]]: "numeric",
    [[minute]]: "numeric",
    [[PatternParts]]: […]
  },
  { [[BiggestDivergentUnit]]: "minute",
    [[hour]]: "numeric",
    [[minute]]: "numeric",
    [[PatternParts]]: […]
  },
  { [[BiggestDivergentUnit]]: ~empty~,
    [[year]]: "2-digit",
    [[month]]: "numeric",
    …,
    [[PatternParts]]: […]
  }
]
```

</details>